services/debugger/transformation-debugger/transformationStatusUploader.go

```diff
@@ -0,0 +1,127 @@
+package transformationdebugger
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/rudderlabs/rudder-server/config"
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+	"github.com/rudderlabs/rudder-server/rruntime"
+	"github.com/rudderlabs/rudder-server/services/debugger"
+	"github.com/rudderlabs/rudder-server/utils"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+//TransformStatusT is a structure to hold transformation status
+type TransformStatusT struct {
+	TransformationID string                 `json:"transformationId"`
+	SourceID         string                 `json:"sourceId"`
+	DestinationID    string                 `json:"destinationId"`
+	EventBefore      map[string]interface{} `json:"eventBefore"`
+	EventAfter       map[string]interface{} `json:"eventAfter"`
+	IsError          bool                   `json:"error"`
+}
+
+type UploadT struct {
+	Payload []interface{} `json:"payload"`
+}
+
+var uploader *debugger.Uploader
+
+var (
+	configBackendURL             string
+	disableTransformationUploads bool
+	pkgLogger                    logger.LoggerI
+)
+
+var uploadEnabledTransformations map[string]bool
+var configSubscriberLock sync.RWMutex
+
+func init() {
+	loadConfig()
+	pkgLogger = logger.NewLogger().Child("debugger").Child("source-debugger")
+}
+
+func loadConfig() {
+	configBackendURL = config.GetEnv("CONFIG_BACKEND_URL", "https://api.rudderlabs.com")
+	disableTransformationUploads = config.GetBool("TransformationDebugger.disableTransformationStatusUploads", false)
+}
+
+type TransformationStatusUploader struct {
+}
+
+func IsUploadEnabled(versionID string) bool {
+	configSubscriberLock.RLock()
+	defer configSubscriberLock.RUnlock()
+	_, ok := uploadEnabledTransformations[versionID]
+	return ok
+}
+
+//Setup initializes this module
+func Setup() {
+	url := fmt.Sprintf("%s/dataplane/eventTransformStatus", configBackendURL)
+	transformationStatusUploader := &TransformationStatusUploader{}
+	uploader = debugger.New(url, transformationStatusUploader)
+	uploader.Setup()
+
+	rruntime.Go(func() {
+		backendConfigSubscriber()
+	})
+}
+
+//RecordTransformationStatus is used to put the transform event in the eventBatchChannel,
+//which will be processed by handleEvents.
+func RecordTransformationStatus(tid, sid, did string, eb, ea map[string]interface{}, iserror bool) bool {
```

Comment:
make the args a struct? since both eb and ea have same type, might make a mistake and send in wrong order and type checker won't complain